Mathlib/Data/Set/Image.lean

```diff
@@ -463,6 +463,10 @@ theorem image_inter_nonempty_iff {f : α → β} {s : Set α} {t : Set β} :
     (f '' s ∩ t).Nonempty ↔ (s ∩ f ⁻¹' t).Nonempty := by
   rw [← image_inter_preimage, image_nonempty]
 
+theorem disjoint_image_left {f : α → β} {s : Set α} {t : Set β} :
+    Disjoint (f '' s) t ↔ Disjoint s (f ⁻¹' t) := by
+  simp_rw [disjoint_iff_inter_eq_empty, ← not_nonempty_iff_eq_empty, image_inter_nonempty_iff]
```

Comment:
We might want a `disjoint_image_right` as well.

Reply:
I agree, then it can be used to golf `Ideal.disjoint_map_primeCompl_iff_comap_le` by defeq. (I no longer need to use this `left` version after I found this `Ideal` lemma.)